fedora-ostree-pruner/fedora-ostree-pruner

```diff
@@ -0,0 +1,253 @@
+                logger.warning('Policy defined for a ref that is not in the '
+                               f' repo {OSTREEPRODREPO}: {ref}')
+            continue
+
+    # prune each branch in the policy with specified value
+    deleted_a_ref = False
+    for ref,policy in PROD_REF_POLICIES.items():
+
+        # Skip if there is no ref in the repo for this policy
+        if ref not in prod_refs:
+            continue
+
+        if policy is None:
+            logger.info(f'Skipping ref {ref} in repo {OSTREEPRODREPO}.'
+                        'Policy is to keep all commits.')
+            continue
+
+        if policy == 'delete':
+            delete_ref_in_repo(ref, OSTREEPRODREPO, test)
+            deleted_a_ref = True
+            continue
+
+        prunearg = pruning_policy_to_args(policy)
+
+        logger.info(f'Pruning the {ref} ref in repo {OSTREEPRODREPO} to {policy}')
+        cmd = ['ostree', 'prune', '--repo', OSTREEPRODREPO,
+               '--only-branch', ref, '--refs-only', prunearg]
+        if test:
+            cmd.append('--no-prune')
+        runcmd(cmd)
+
+    # If we deleted a ref from the repo then we'll just run a generic
+    # prune to delete any objects in the repo that aren't reachable
+    # via a ref. This should save some time since we only call ostree
+    # prune once, versus for every ref we deleted.
+    if deleted_a_ref:
+        cmd = ['ostree', 'prune', '--repo', OSTREEPRODREPO, '--refs-only']
+        if test:
+            cmd.append('--no-prune')
+        runcmd(cmd)
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--test",
+                        help="Don't actually prune", action='store_true')
+    parser.add_argument("--loop",
+                        help="Loop forever, run once a week.", action='store_true')
+    args = parser.parse_args()
+
+    # set args.test = True for now. Once we have proved
+    # everything out in prod and the logs look good we'll
+    # delete this code.
+    args.test = True
+
+    # on startup let's print out the config that was generated
+    logger.info('The configured policy is: \n%s' %
+                pprint.pformat(PROD_REF_POLICIES, indent=8))
+
+    # Configure a umask of 0002 which will allow for the group permissions
+    # to include write for newly created files. We need this because we'd
+    # like to access the OSTree repos from two different Kubernetes pods,
+    # which will have different UIDs but the same GID.
+    # See https://pagure.io/releng/issue/8811#comment-616490
+    os.umask(0o0002)
+
+    # Prune the compose and the prod repo based on policy
+    prune_compose_repo(args.test)
+    prune_prod_repo(args.test)
+
+    # If we were asked to run in a loop, then run once a week on
+    # Saturday.
+    if args.loop:
+        while True:
+            # If it's Saturday, run the prune. Otherwise sleep
+            weekday = datetime.date.today().weekday()
+            if weekday == 5:
```

Comment:
The 5 is kind of opaque. There's `time.strftime("%A")` though that's locale-dependent I think. Maybe we could just have a list like:

```
weekdays = ['Monday', ...]
```

And then we could do

```
weekday = weekdays[datetime.date.today().weekday()]
if weekday == 'Saturday':
  ...
```

And in the log message below we could say e.g. `Today is {weekday}. Sleeping...`.

Reply:
Ok I updated it in the next upload to be closer to what you suggested.